pep-0621.rst

```diff
@@ -81,7 +85,17 @@ then tools MUST raise an error to notify the user about their mistake.
 Data specified using this PEP is considered canonical. Tools CANNOT
 remove or change data, but they MAY add to it. This allows for tools
 to make data more accurate/static when possible by updating the data
-specified in the ``pyproject.toml`` file.
+specified in the ``pyproject.toml`` file. For example, a version
+can become more specific when building a wheel (e.g. adding a local
+version), but it cannot become less specific.
+
+Build back-ends creating a source distribution -- aka an "sdist" --
+SHOULD provide as much data as possible using this PEP within a source
+distribution. Some fields specified below MUST be provided statically
+in a source distribution *if* the metadata is to be provided at any
+point for the project. Other metadata must *always* be provided
```

Comment:
I'm not sure what "if the metadata is to be provided at any point for the project" means here.

I think what it means is that once you have a sdist, these fields must be removed from the "dynamic" list - so that they are either guaranteed static, or guaranteed not to be specified. So the only difference between these and "version" is that "version" cannot be omitted.

In which case, I'd make `dynamic` the key thing here, and say that "Some fields specified below MAY NOT be included in the `dynamic` list in a source distribution, implying that the value (or lack of value) in the sdist's `pyproject.toml` is guaranteed to be the correct value for any build of the project". (I'm not particularly happy with the second part of that sentence, after "implying that..." so feel free to come up with a better wording 🙂)

Then just add a note that "name" and "version" CANNOT be omitted from a sdist.

Reply:
Yep, that's exactly what I'm getting after with that phrase. I'll tweak the wording.